src/include/mpir_pset.h

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (C) by Argonne National Laboratory
+ *     See COPYRIGHT in top-level directory
+ */
+
+#include "mpiimpl.h"
+
+/**
+ * @brief Process set structure
+ */
+struct MPIR_Pset {
+    char *uri;                  /* Name of the process set */
+    int size;                   /* Number of processes in the set */
+    bool is_valid;              /* True if process set is valid, false if not */
+    int *members;               /* Array of members of the process set (ranks) sorted in ascending order */
+};
+
+/**
+ * @brief Process set array structure
+ */
+struct MPIR_Pset_array {
+    MPIR_OBJECT_HEADER;
+    UT_array *parray;           /* Array of psets */
+    MPID_Thread_mutex_t *mutex; /* Optional mutex to protect parray from concurrent accesses */
+};
```

Comment:
Would it be sufficient to just use a single global UT_array for all psets?

Reply:
We agreed offline to remove the array structures and the PR has been revised.